test/utility/test_util.i

```diff
@@ -0,0 +1,133 @@
+/*-
+ * Public Domain 2014-2015 MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <wiredtiger.h>
+
+#ifdef _WIN32
+#include "windows_shim.h"
+#endif
+
+#ifdef _WIN32
+	#define DIR_DELIM '\\'
+#else
+	#define	DIR_DELIM '/'
+#endif
+
+/*
+ * die --
+ *	Report an error and quit.
+ */
+static inline void
+testutil_die(int e, const char *fmt, ...)
+{
+	va_list ap;
+
+	va_start(ap, fmt);
+	vfprintf(stderr, fmt, ap);
+	va_end(ap);
+	if (e != 0)
+		fprintf(stderr, ": %s", wiredtiger_strerror(e));
+	fprintf(stderr, "\n");
+	exit(EXIT_FAILURE);
+}
+
+/*
+ * testutil_work_dir_from_path --
+ *	Takes a buffer, its size and the intended work directory.
+ *	Creates the full intended work directory in buffer.
+ */
+static inline int
+testutil_work_dir_from_path(char *buffer, size_t inputSize, char *dir)
+{
+	/* If no directory is provided, use the default. */
+	if (dir == NULL) {
+		if (inputSize < 9)
+			return (1);
+		snprintf(buffer, inputSize, "WT_TEST");
+		return (0);
+	}
+
+	/* 9 bytes for the directory and WT_TEST */
+	if (inputSize < strlen(dir) + 9)
+		return (1);
+
+	/* Is this windows or *nix? */
+#ifdef _WIN32
+	snprintf(buffer, inputSize, "%s\\WT_TEST", dir);
+#else
+	snprintf(buffer, inputSize, "%s/WT_TEST", dir);
+#endif
+	return (0);
+}
+
+/*
+ * testutil_clean_work_dir --
+ *	Remove any existing work directories, can optionally fail on error
+ */
+static inline int
+testutil_clean_work_dir(char *dir)
+{
+	char *buffer;
+	size_t inputSize;
+
+	/* 10 bytes for the Windows rd command */
+	inputSize = strlen(dir) + 19;
+	buffer = (char*) malloc (inputSize);
+
+#ifdef _WIN32
+	snprintf(buffer, inputSize, "rd /s /q %s", dir);
```

Comment:
You will have to check style with @agorrod and @michaelcahill, but I would recommend a different pattern. Instead of `#ifdef`s wrapping code, wrap the constant string like on you did on line 37. Then to get the length of the string, you can just use `sizeof()`, and can avoid hard coded numbers like 9, 10, and 19.